jobwatcher/jobwatcher.py

```diff
@@ -64,61 +83,94 @@ def read_cfnconfig():
             log.info("Instance %s has %s vcpus." % (instance_type, vcpus))
-            return vcpus
-        except KeyError as e:
-            log.error("Instance %s not found in file %s." % (instance_type, pricing_file))
-            exit(1)
+        except KeyError:
+            log.error("Unable to get vcpus from file %s. Instance type %s not found." % (pricing_file, instance_type))
+            vcpus = -1
 
+        return vcpus
 
-def get_instance_properties(instance_type):
-    cfnconfig_params = read_cfnconfig()
+
+def _get_instance_properties(instance_type):
+    """
+    Get instance properties for the given instance type, according to the cfn_scheduler_slots configuration parameter.
+
+    :param instance_type: instance type to search for
+    :return: a dictionary containing the instance properties. E.g. {'slots': <slots>}
+    """
     try:
+        cfnconfig_params = _read_cfnconfig()
         cfn_scheduler_slots = cfnconfig_params["cfn_scheduler_slots"]
-        slots = 0
-        vcpus = get_vcpus_from_pricing_file(instance_type)
-
-        if cfn_scheduler_slots == "cores":
-            log.info("Instance %s will use number of cores as slots based on configuration." % instance_type)
-            slots = -(-vcpus//2)
-        elif cfn_scheduler_slots == "vcpus":
-            log.info("Instance %s will use number of vcpus as slots based on configuration." % instance_type)
+    except KeyError:
+        log.error(
+            "Required config parameter 'cfn_scheduler_slots' not found in file %s. Assuming 'vcpus'" % cfnconfig_file
+        )
+        cfn_scheduler_slots = "vcpus"
+
+    vcpus = _get_vcpus_from_pricing_file(instance_type)
+
+    if cfn_scheduler_slots == "cores":
+        log.info("Instance %s will use number of cores as slots based on configuration." % instance_type)
+        slots = -(-vcpus//2)
+
+    elif cfn_scheduler_slots == "vcpus":
+        log.info("Instance %s will use number of vcpus as slots based on configuration." % instance_type)
+        slots = vcpus
+
+    elif cfn_scheduler_slots.isdigit():
+        slots = int(cfn_scheduler_slots)
+        log.info("Instance %s will use %s slots based on configuration." % (instance_type, slots))
+
+        if slots <= 0:
+            log.error(
+                "cfn_scheduler_slots config parameter '%s' must be greater than 0. "
+                "Assuming 'vcpus'" % cfn_scheduler_slots
+            )
             slots = vcpus
-        elif cfn_scheduler_slots.isdigit():
-            slots = int(cfn_scheduler_slots)
-            log.info("Instance %s will use %s slots based on configuration." % (instance_type, slots))
+    else:
+        log.error("cfn_scheduler_slots config parameter '%s' is invalid. Assuming 'vcpus'" % cfn_scheduler_slots)
+        slots = vcpus
 
-        if not slots > 0:
-            log.critical("cfn_scheduler_slots config parameter '%s' was invalid" % cfn_scheduler_slots)
-            exit(1)
+    if slots <= 0:
```

Comment:
minor: you can just check < since you are going to set it to 0 in any case.

Reply:
done